cloudevents/http/util.py

```diff
@@ -25,10 +25,21 @@ def default_marshaller(content: any):
         return content
 
 
-def _json_or_string(content: typing.Union[str, bytes]):
+ContentT = typing.TypeVar("ContentT", bound=typing.Union[str, bytes])
+
+
+def _json_or_string(
+    content: typing.Optional[ContentT],
+) -> typing.Optional[
+    typing.Union[
+        typing.Dict[typing.Any, typing.Any],
+        typing.List[typing.Any],
+        ContentT,
+    ]
+]:
```

Comment:
Let's use standard `AnyStr` instead of custom `ContentT`.
```suggestion
def _json_or_string(
    content: typing.Optional[typing.AnyStr],
) -> typing.Optional[
    typing.Union[
        typing.Dict[typing.Any, typing.Any],
        typing.List[typing.Any],
        typing.AnyStr,
    ]
]:
```